auth-service/src/main/kotlin/uk/gov/communities/delta/auth/Routing.kt

```diff
@@ -328,7 +329,29 @@ fun Route.bearerTokenRoutes(
     }
 }
 
-fun Route.serviceUserRoutes(samlTokenController: GenerateSAMLTokenController) {
+fun Route.serviceUserRoutes(
+    samlTokenController: GenerateSAMLTokenController,
+    editLdapGroupsController: EditLdapGroupsController
+) {
+    samlTokenRoutes(samlTokenController)
+
+    authenticate(DELTA_AD_LDAP_SERVICE_USERS_AUTH_NAME, strategy = AuthenticationStrategy.Required) {
```

Comment:
Can use the client header auth too (i.e. authenticating which service is making the call) if we send it from MarkLogic
```
map:entry("Delta-Client", $config:SAML-TOKEN-GENERATOR-CLIENT-ID || ":" || $config:SAML-TOKEN-GENERATOR-CLIENT-SECRET),
```
in the header map like saml-token.xqy. Doesn't really matter here since we're only expecting MarkLogic to call it, up to you.

Reply:
I think just basic auth is more intuitive (and less work)